pyro/ops/studentt.py

```diff
@@ -0,0 +1,387 @@
+    """
+    def __init__(self, log_normalizer, info_vec, precision, alpha, beta):
+        # NB: using info_vec instead of mean to deal with rank-deficient problem
+        assert info_vec.dim() >= 1
+        assert precision.dim() >= 2
+        assert precision.shape[-2:] == info_vec.shape[-1:] * 2
+        self.log_normalizer = log_normalizer
+        self.info_vec = info_vec
+        self.precision = precision
+        self.alpha = alpha
+        self.beta = beta
+
+    def dim(self):
+        return self.info_vec.size(-1)
+
+    @lazy_property
+    def batch_shape(self):
+        return broadcast_shape(self.log_normalizer.shape,
+                               self.info_vec.shape[:-1],
+                               self.precision.shape[:-2],
+                               self.alpha.shape,
+                               self.beta.shape)
+
+    def expand(self, batch_shape):
+        n = self.dim()
+        log_normalizer = self.log_normalizer.expand(batch_shape)
+        info_vec = self.info_vec.expand(batch_shape + (n,))
+        precision = self.precision.expand(batch_shape + (n, n))
+        alpha = self.alpha.expand(batch_shape)
+        beta = self.beta.expand(batch_shape)
+        return GaussianGamma(log_normalizer, info_vec, precision, alpha, beta)
+
+    def reshape(self, batch_shape):
+        n = self.dim()
+        log_normalizer = self.log_normalizer.reshape(batch_shape)
+        info_vec = self.info_vec.reshape(batch_shape + (n,))
+        precision = self.precision.reshape(batch_shape + (n, n))
+        alpha = self.alpha.reshape(batch_shape)
+        beta = self.beta.reshape(batch_shape)
+        return GaussianGamma(log_normalizer, info_vec, precision, alpha, beta)
+
+    def __getitem__(self, index):
+        """
+        Index into the batch_shape of a GaussianGamma.
+        """
+        assert isinstance(index, tuple)
+        log_normalizer = self.log_normalizer[index]
+        info_vec = self.info_vec[index + (slice(None),)]
+        precision = self.precision[index + (slice(None), slice(None))]
+        alpha = self.alpha[index]
+        beta = self.beta[index]
+        return GaussianGamma(log_normalizer, info_vec, precision, alpha, beta)
+
+    @staticmethod
+    def cat(parts, dim=0):
+        """
+        Concatenate a list of GaussianGammas along a given batch dimension.
+        """
+        if dim < 0:
+            dim += len(parts[0].batch_shape)
+        args = [torch.cat([getattr(g, attr) for g in parts], dim=dim)
+                for attr in ["log_normalizer", "info_vec", "precision", "alpha", "beta"]]
+        return GaussianGamma(*args)
+
+    def event_pad(self, left=0, right=0):
+        """
+        Pad along event dimension.
+        """
+        lr = (left, right)
+        info_vec = pad(self.info_vec, lr)
+        precision = pad(self.precision, lr + lr)
+        # no change for alpha, beta because we are working with reparameterized version
```

Comment:
Would it be more precise to say

> no change for log_normalizer, alpha, and beta because they are shared across dimensions

Reply:
I just add a comment to clarify this point.